### PR TITLE
GODRIVER-3249: Handle all possible OIDC configuration errors

### PR DESCRIPTION
**Pull request by pmeredit**
GODRIVER-3249
## Summary

Add configuration errors for all malformed configuration for the OIDC authentication mechanism. Some of these are redundant with errors thrown by the OIDC code itself, but these should occur before that point. I think double checking is fine.

## Background & Motivation

OIDC epic

### CODE REVIEW

File: x/mongo/driver/auth/internal/gssapi/gss.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ package gssapi
 */
```
Comment (pmeredit) [inline x/mongo/driver/auth/internal/gssapi/gss.go:19]: I'm not sure why these gss changes are getting pulled in, these are already in v1 from our final merge. Very confusing

File: x/mongo/driver/auth/oidc.go [hunk 2]
```diff
@@ -13,29 +13,27 @@ import (
 	"io"
 	"net/http"
```
Comment (pmeredit) [inline x/mongo/driver/auth/oidc.go:14]: Again we are seeing some changes from the end of the human flow PR here. The relevant changes are just exporting some of these values to make use of them in configuration errors.

**Review by pmeredit — COMMENTED** [review]
I'm not sure why the PR is picking up changes that are in v1. I've tried rerooting this.

File: x/mongo/driver/auth/oidc.go [hunk 3]
```diff
@@ -26,14 +26,24 @@ import (
 // MongoDBOIDC is the string constant for the MONGODB-OIDC authentication mechanism.
```
Comment (pmeredit) [inline x/mongo/driver/auth/oidc.go:26]: These are now public so they can be reused in configuration error checking to keep consistency if we ever need to change the values.

**Comment by pmeredit** [comment]
I added in [GODRIVER-3226](https://jira.mongodb.org/browse/GODRIVER-3226) because the GODRIVER was actually handling this correctly, unlike the python driver, and all I needed was to add tests showing that.

File: mongo/options/clientoptions.go [hunk 4]
```diff
@@ -360,6 +361,36 @@ func (c *ClientOptions) validate() error {
...
+			return fmt.Errorf("password must not be set for the %s auth mechanism", auth.MongoDBOIDC)
+		}
+		if c.Auth.OIDCMachineCallback != nil && c.Auth.OIDCHumanCallback != nil {
+			return fmt.Errorf("cannot set both OIDCMachineCallback and OIDCHumanCallback, only one may be specified")
+		}
+		if env, ok := c.Auth.AuthMechanismProperties[auth.EnvironmentProp]; ok {
+			switch env {
+			case auth.GCPEnvironmentValue:
+				fallthrough
+			case auth.AzureEnvironmentValue:
```
Comment (matthewdale) [inline mongo/options/clientoptions.go:376]: Optional: Consider combining these cases.

```suggestion
			case auth.GCPEnvironmentValue, auth.AzureEnvironmentValue:
```

**Review by matthewdale — APPROVED** [review]
Looks good! 👍

File: mongo/options/clientoptions.go (hunk 4 above)
Comment (pmeredit) [reply mongo/options/clientoptions.go:376]: I forgot this was possible :joy: thanks!

**Pull request merged**